### PR TITLE
Update to kubernetes-1.15.4 and helm 2.14.3

### PR DESCRIPTION
**Pull request by sslavic**
Towards https://github.com/giantswarm/giantswarm/issues/6302

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -55,7 +55,7 @@ required = [
   name = "github.com/giantswarm/errors"
 
 [[constraint]]
-  branch = "master"
+  branch = "kubernetes-1.15.4"
```
Comment (sslavic) [inline Gopkg.toml]: To be reverted once https://github.com/giantswarm/k8sportforward/pull/19 gets merged.

File: integration/test/upgradetiller/upgrade_tiller_test.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ func TestUpgradeTiller(t *testing.T) {
 			t.Fatalf("could not get tiller image %#v", err)
 		}
 		if tillerImage != currentTillerImage {
-			t.Fatalf("tiller image == %#q, want %#q", tillerImage, outdatedTillerImage)
+			t.Fatalf("tiller image == %#q, want %#q", tillerImage, currentTillerImage)
```
Comment (sslavic) [inline integration/test/upgradetiller/upgrade_tiller_test.go]: Uncovered this (likely copy/paste) bug in error message - "current" was used for comparison, but "outdated" was used in the message.

File: integration/test/upgradetiller/upgrade_tiller_test.go [hunk 3]
```diff
@@ -80,7 +80,7 @@ func getTillerDeployment(ctx context.Context, namespace string, labelSelector st
 			lo := metav1.ListOptions{
 				LabelSelector: labelSelector,
 			}
-			l, err := config.CPK8sClients.K8sClient().Apps().Deployments(namespace).List(lo)
+			l, err := config.CPK8sClients.K8sClient().AppsV1().Deployments(namespace).List(lo)
```
Comment (sslavic) [inline integration/test/upgradetiller/upgrade_tiller_test.go]: Deprecated versionless API was removed via https://github.com/kubernetes/client-go/commit/d2861e956fe1b2303c7a1688339f4aa2136c8c76#diff-76b263f1379ed2a72308d69111a53315

File: integration/test/upgradetiller/upgrade_tiller_test.go (hunk 2 above)
Comment (rossf7) [reply integration/test/upgradetiller/upgrade_tiller_test.go]: Good catch, thx!

**Pull request merged**